<?xml version="1.0" ?>
<launch>
    <arg name="eye_on_hand" doc="eye-on-hand instead of eye-on-base" />
    <arg name="namespace_prefix" default="easy_handeye" />
    <arg if="$(arg eye_on_hand)" name="namespace" value="$(arg namespace_prefix)_eye_on_hand" />
    <arg unless="$(arg eye_on_hand)" name="namespace" value="$(arg namespace_prefix)_eye_on_base" />

    <!-- The input reference frames -->
    <arg name="robot_base_frame" default="base_link" />
    <arg name="robot_effector_frame" default="tool0" />
    <arg name="tracking_base_frame" default="optical_origin" />
    <arg name="tracking_marker_frame" default="optical_target" />

    <!-- Publish frames while calibrating -->
    <arg name="publish_tf" default="true"/>

    <!-- Run the ViSP solver to determine the extrinsic parameters -->
    <node ns="$(arg namespace)" name="hand_eye_solver"
          pkg="visp_hand2eye_calibration"
          type="visp_hand2eye_calibration_calibrator">
        <remap from="/compute_effector_camera_quick" to="compute_effector_camera_quick"/>
    </node>

    <group ns="$(arg namespace)" >
        <param name="eye_on_hand" value="$(arg eye_on_hand)"/>
        <param name="robot_base_frame" value="$(arg robot_base_frame)" />
        <param name="robot_effector_frame" value="$(arg robot_effector_frame)" />
        <param name="tracking_base_frame" value="$(arg tracking_base_frame)" />
        <param name="tracking_marker_frame" value="$(arg tracking_marker_frame)" />
    </group>

    <!-- Connect the tracker to the solver -->
    <node name="hand_eye_connector"
          ns="$(arg namespace)"
          pkg="easy_handeye"
          type="calibrate.py"
          output="screen" />

</launch>
